### PR TITLE
Add Java section, and update README

### PR DESCRIPTION
**Pull request by nomisRev**

### CODE REVIEW

**Review by Yawolf — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -90,16 +90,36 @@ libraryDependencies += "com.xebia" %% "xef-scala" % "<version>"
...
+- [<img src="https://en.wikipedia.org/wiki/Java_(programming_language)#/media/File:Java_programming_language_logo.svg" height="15px" alt="Scala logo"> Java version](https://github.com/xebia-functional/xef/blob/main/docs/intro/java.md)
 
 ## 🚀 Examples
 
 You can also have a look at the examples to have a feeling of how using the library looks like.
 
 - [<img src="https://upload.wikimedia.org/wikipedia/commons/3/37/Kotlin_Icon_2021.svg" height="15px" alt="Kotlin logo"> Examples in Kotlin](https://github.com/xebia-functional/xef/tree/main/examples/kotlin/src/main/kotlin/com/xebia/functional/xef/auto)
 - [<img src="https://www.scala-lang.org/resources/img/frontpage/scala-spiral.png" height="15px" alt="Scala logo"> Examples in Scala](https://github.com/xebia-functional/xef/tree/main/examples/scala/src/main/scala/com/xebia/functional/xef/scala/auto)
+- [<img src="https://en.wikipedia.org/wiki/Java_(programming_language)#/media/File:Java_programming_language_logo.svg" height="15px" alt="Scala logo"> Examples in Java](https://github.com/xebia-functional/xef/tree/main/examples/java/src/main/java/com/xebia/functional/xef/java/auto)
```
Comment (anamariamv) [inline README.md:125]:
```suggestion
- [<img src="https://en.wikipedia.org/wiki/Java_(programming_language)#/media/File:Java_programming_language_logo.svg" height="15px" alt="Java logo"> Examples in Java](https://github.com/xebia-functional/xef/tree/main/examples/java/src/main/java/com/xebia/functional/xef/java/auto)
```

File: README.md [hunk 2]
```diff
@@ -90,16 +90,36 @@ libraryDependencies += "com.xebia" %% "xef-scala" % "<version>"
...
+
+</details>
+
 ## 📖 Quick Introduction
 
 In this small introduction we look at the main features of xef, including the `ai` function.
 
 - [<img src="https://upload.wikimedia.org/wikipedia/commons/3/37/Kotlin_Icon_2021.svg" height="15px" alt="Kotlin logo"> Kotlin version](https://github.com/xebia-functional/xef/blob/main/docs/intro/kotlin.md)
 - [<img src="https://www.scala-lang.org/resources/img/frontpage/scala-spiral.png" height="15px" alt="Scala logo"> Scala version](https://github.com/xebia-functional/xef/blob/main/docs/intro/scala.md)
+- [<img src="https://en.wikipedia.org/wiki/Java_(programming_language)#/media/File:Java_programming_language_logo.svg" height="15px" alt="Scala logo"> Java version](https://github.com/xebia-functional/xef/blob/main/docs/intro/java.md)
```
Comment (anamariamv) [inline README.md:117]:
```suggestion
- [<img src="https://en.wikipedia.org/wiki/Java_(programming_language)#/media/File:Java_programming_language_logo.svg" height="15px" alt="Java logo"> Java version](https://github.com/xebia-functional/xef/blob/main/docs/intro/java.md)
```

**Review by anamariamv — APPROVED** [review]
Minor typo

File: README.md [hunk 3]
```diff
@@ -90,16 +90,36 @@ libraryDependencies += "com.xebia" %% "xef-scala" % "<version>"
...
+<details>
+<summary><img src="https://en.wikipedia.org/wiki/Apache_Maven#/media/File:Apache_Maven_logo.svg" height="15px" alt="Maven logo"> Maven</summary>
+
+Libraries are published in Maven Central. You may need to  add that repository explicitly
+in your build, if you haven't done it before.
+
+```xml
+<dependency>
+   <groupId>com.xebia</groupId>
+   <artifactId>xef-core</artifactId>
```
Comment (juanpedromoreno) [inline README.md:102]:
```suggestion
   <artifactId>xef-java</artifactId>
```

**Review by juanpedromoreno — CHANGES_REQUESTED** [review]
Thanks @nomisRev , I think we need to remove the `xef-core` dependency from the docs everywhere.

**Comment by nomisRev** [comment]
You're absolutely right @juanpedromoreno. I removed all references to `xef-core` from the documentation with exception of the badges.

**Review by juanpedromoreno — APPROVED** [review]
🙌

**Pull request merged**